rviz_default_plugins/src/rviz_default_plugins/transformation/tf_wrapper.cpp

```diff
@@ -123,8 +123,13 @@ void TFWrapper::initialize(
   bool using_dedicated_thread)
 {
   initializeBuffer(clock, rviz_ros_node.lock()->get_raw_node(), using_dedicated_thread);
-  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(
-    *buffer_, rviz_ros_node.lock()->get_raw_node(), false);
+  if (using_dedicated_thread) {
+    tf_listener_ = std::make_shared<tf2_ros::TransformListener>(
+      *buffer_, true);
```

Comment:
This will actually create a new node, did you try what happens if you just use `std::make_shared<tf2_ros::TransformListener>(
    *buffer_, rviz_ros_node.lock()->get_raw_node(), true);` - shouldn't this already spin the TransformListener in a dedicated thread?

Reply:
`Node has already been added to an executor.` was shown.
relevant code:
- https://github.com/ros2/rclcpp/blob/master/rclcpp/src/rclcpp/executor.cpp#L122
- https://github.com/ros2/rviz/blob/ros2/rviz_common/src/rviz_common/visualization_manager.cpp#L236
- https://github.com/ros2/geometry2/blob/ros2/tf2_ros/src/transform_listener.cpp#L78

Reply:
Okay, I see... So the correct way to solve this should be to actually make RViz multi-threaded and spin the transformer callback in its own thread (and more often than e.g. the update loop). 

I guess this is a much more subtle undertaking, so I would propose:
- Add a comment that this actually introduces a new node and should be done via threading,
- Merge the PR since this fixes a big issue where there are multiple reports
- Open an issue (or use this one: https://github.com/ros2/rviz/issues/548) to track the todo

Reply:
> Add a "todo" comment that this actually introduces a new node and should be done via threading,

If you can provide the concrete comment, I'll add them to this PR. (I don't know the format of the TODO comments on this project.)

Reply:
No real schema except for linters. The following should do (you can also add your name, but I find the naming inconsistent, it's just enforced by cpplint)
```
TODO(anyone): The TransformListener needs very quick spinning so it uses its own node here. Remove this in favor of a multithreaded spinner and ensure that the listener callback queue does not fill up.
```
You'll probably need to split it up into multiple lines.

@jacobperron The error analysis seems sound and the solution works on several machines. When this got ported we explicitly avoided making new nodes if possible, but making RViz multithreaded requires much more work, since this is an issue many people complained about, I think this is a good compromise - unless @ymd-stella wants to jump in and do this of course.
Maybe you could take a quick look?

Reply:
I agree that a multithreaded executor is sounds like the best way to go. I'm okay with merging this workaround in the meantime with a TODO as you suggest.

Reply:
I've added the comment. I've written `TODO(pull/551)` to make it easier for those who found it to look up the details.